src/windows/BluetoothLePlugin.js

```diff
@@ -225,62 +226,64 @@ module.exports = {
       return;
     }
 
-    var DeviceInformation = Windows.Devices.Enumeration.DeviceInformation;
-    var DeviceInformationKind = Windows.Devices.Enumeration.DeviceInformationKind;
-
-    WinJS.Promise.wrap(address)
-    .then(function (deviceAddress) {
-      // If we have cached device info return it right now
-      if (WATCH_CACHE[deviceAddress]) return [WATCH_CACHE[deviceAddress]];
-      // Otherwise try to search it again
-      var selector = "System.Devices.Aep.ProtocolId:=\"{bb7bb05e-5972-42b5-94fc-76eaa7084d49}\" AND " +
-                      "System.Devices.Aep.ContainerId:=\"{" + deviceAddress + "}\" AND " +
-                      "(System.Devices.Aep.CanPair:=System.StructuredQueryType.Boolean#True OR " +
-                      "System.Devices.Aep.IsPaired:=System.StructuredQueryType.Boolean#True)";
-      return DeviceInformation.findAllAsync(selector, ["System.Devices.Aep.ContainerId"], DeviceInformationKind.associationEndpoint);
-    })
-    .then(function (devices) {
-      return Windows.Devices.Bluetooth.BluetoothLEDevice.fromIdAsync(devices[0].id);
-    })
-    .then(function (bleDevice) {
-      var DevicePairingProtectionLevel = Windows.Devices.Enumeration.DevicePairingProtectionLevel;
-      var DevicePairingResultStatus = Windows.Devices.Enumeration.DevicePairingResultStatus;
-
-      if (bleDevice.deviceInformation.pairing.isPaired) {
+    getDeviceByAddress(address)
+    // .then(function (bleDevice) {
```

Comment:
Did this get a lot simpler? Remove all the commented out / old code

Reply:
Yes, I was able to simplify it a ton. I'll remove the cruft.